### ISSUE TITLE
Paginators, pages do no support embed fields

### ISSUE THREAD

**Issue by Yofou** [Kotlin-Discord/kord-extensions#67]
# Summary

The Title is basically the summary, so not much to add here.
but I would love to be able to add some fields to an embed through
the Page API, so I can use the Paginator utility more.

# Current Behaviour

The current page builder only contains this..
```kt
public open class Page(
    public open val description: String,
    public open val title: String? = null,
    public open val author: String? = null,
    public open val authorIcon: String? = null,
    public open val authorUrl: String? = null,
    public open val color: Color? = null,
    public open val footer: String? = null,
    public open val footerIcon: String? = null,
    public open val image: String? = null,
    public open val thumbnail: String? = null,
    public open val url: String? = null,
    public open val bundle: String? = null,
)
```

as you can see, fields are absent in the constructor (and I doubled checked for any methods too),
which I feel are an essential feature of they embed in general

# Preferred Behaviour

if it was up to me, I think it would be a lot nice to just be able to hook into the embed builder in general,
rather than making this kind of makeshift builder class, so it would be used like this
```kt
val paginator = paginator {
    page {
        title = "I an embed title"
        description = "I am description"

       field {
          name = "i am field title"
          value = "i am field value"
       }
    }   
}
```

# Suggestions

depends on how you continue how the paginator behaves, 
but I believe the paginator builder is a lot better DX than they 
class constructor currently

**Comment by gdude2002** [comment]
I agree that there should be a builder. Ultimately, one needs to be able to provide a bundle - for translation purposes. It's also worth noting that pages are _not_ "plain" embed builders, and do modify the embeds as well.

It may still be possible to just use an embed builder, though. I'll think about it.

**Comment by gdude2002** [comment]
This has been done and will be available when `1.4.4-RC3` releases.
